src/main/scala/com.snowplowanalytics/forex/ForexClient.scala

```diff
@@ -29,46 +35,40 @@ import com.twitter.util.SynchronizedLruMap
 object ForexClient {
 
   /**
-   * Getter for clients with specified caches(optional)
+   * Getter for clients, creating the caches as defined in the config
    */
-  def getClient(config: ForexConfig,
-                clientConfig: ForexClientConfig,
-                nowish: MaybeNowishCache = None,
-                eod: MaybeEodCache       = None): ForexClient =
+  def getClient[F[_]: Sync](config: ForexConfig, clientConfig: ForexClientConfig): F[ForexClient[F]] =
     clientConfig match {
       case oerClientConfig: OerClientConfig =>
-        new OerClient(config, oerClientConfig, nowishCache = nowish, eodCache = eod)
-      case _ => throw NoSuchClientException("This client is not supported by scala-forex currently")
-    }
-}
+        val nowishCacheF =
+          if (config.nowishCacheSize > 0)
+            LruMap.create[F, NowishCacheKey, NowishCacheValue](config.nowishCacheSize).map(_.some)
+          else Sync[F].pure(Option.empty[NowishCache[F]])
 
-abstract class ForexClient(config: ForexConfig, nowishCache: MaybeNowishCache = None, eodCache: MaybeEodCache = None) {
+        val eodCacheF =
+          if (config.eodCacheSize > 0)
+            LruMap.create[F, EodCacheKey, EodCacheValue](config.eodCacheSize).map(_.some)
+          else Sync[F].pure(Option.empty[EodCache[F]])
 
-  // Assemble our caches
-  object caches {
-
-    // LRU cache for nowish request, with (source currency, target currency) as the key
-    // and (date time, exchange rate) as the value
-    val nowish =
-      if (nowishCache.isDefined) {
-        nowishCache
-      } else if (config.nowishCacheSize > 0) {
-        Some(new SynchronizedLruMap[NowishCacheKey, NowishCacheValue](config.nowishCacheSize))
-      } else {
-        None
-      }
+        (nowishCacheF, eodCacheF).mapN {
+          case (nowish, eod) =>
+            new OerClient[F](config, oerClientConfig, nowishCache = nowish, eodCache = eod)
+        }
+      case _ => throw NoSuchClientException("This client is not supported by scala-forex currently")
+    }
 
-    // LRU cache for historical request, with (source currency, target currency, date time) as the key
-    // and exchange rate as the value
-    val eod =
-      if (eodCache.isDefined) {
-        eodCache
-      } else if (config.eodCacheSize > 0) {
-        Some(new SynchronizedLruMap[EodCacheKey, EodCacheValue](config.eodCacheSize))
-      } else {
-        None
-      }
+  def getClient[F[_]: Sync](config: ForexConfig,
+                            clientConfig: ForexClientConfig,
+                            nowishCache: Option[NowishCache[F]],
+                            eodCache: Option[EodCache[F]]): ForexClient[F] = clientConfig match {
```

Comment:
`ForexClientConfig` is a trait without any members. `OerClientConfig` is its only known subtype. Basically it means that anything can extend `ForexClientConfig`, but nothing in this library can benefit from it? I guess if we'll unify them - we could also get rid of runtime exceptions.